### ISSUE TITLE
Linear regression gradient/hessian

### ISSUE THREAD

**Issue by eckelsjd** [eckelsjd/amisc#48]
Need to implement the gradient and Hessian functions of the `Linear(Interpolator)` class.

Can probably be done analytically by pulling out all of the polynomial features and coefficients from sklearn. But maybe they already have a built-in way to compute this.

**Comment by eckelsjd** [comment]
Also need to implement gradient and Hessian for the `GPR(Interpolator)` class. See #49
